spec/factories/supervisor_reports.rb

```diff
@@ -20,35 +17,28 @@
 
     before :create do |report|
       report.saved_pictures = rand 50 if report.pictures_saved?
-
       if rand(20).zero?
         report.amplifying_comments = FFaker::Lorem.paragraph
       end
 
       rand(3).times do
         create :witness, supervisor_report: report
       end
-
-      if report.completed_drug_or_alcohol_test?
-        case report.reason_test_completed
-        when 'Post-Accident'
-          field = %w[
-            test_due_to_bodily_injury
-            test_due_to_disabling_damage
-            test_due_to_fatality
-          ].sample
-          report.send field + '=', true
-        when 'Reasonable Suspicion'
-          report.observation_made_at = Time.zone.now - 5.minutes
-          reason = %w[appearance behavior speech odor].sample
-          report.send "test_due_to_employee_#{reason}=", true
-          report.send "employee_#{reason}=", FFaker::Lorem.sentence
-        end
-      elsif [true, false].sample
-        report.fta_threshold_not_met = true
+      if report.test_status.include?('Post-Accident: Threshold met (completed drug test)')
+        field = %w[
+          test_due_to_bodily_injury
+          test_due_to_disabling_damage
+          test_due_to_fatality
+        ].sample
```

Comment:
Again let's stop using sample in factories so it's not random.

Reply:
I don't think any of these options have any validations?

Reply:
The factory should contain only what is necessary for an object to exist. Anything extra should be contained in the specific test that relies on it.

Reply:
Aight, done.